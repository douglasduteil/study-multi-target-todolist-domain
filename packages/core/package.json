{
  "name": "@todolist/core",
  "description": "The core todolist system",
  "version": "1.0.0",
  "private": true,
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "scripts": {
    "build": "tsc",
    "lint": "tslint -p tsconfig.json -t stylish",
    "test": "jest",
    "watch": "yarn build --watch --preserveWatchOutput"
  },
  "dependencies": {
    "ramda": "^0.26.1"
  },
  "devDependencies": {
    "@types/jest": "^23.3.12",
    "@types/ramda": "^0.25.46",
    "jest": "^23.6.0",
    "ts-jest": "^23.10.5",
    "tslint": "^5.12.1",
    "typescript": "^3.2.2"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/douglasduteil/study-multi-target-todolist-domain.git"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/douglasduteil/study-multi-target-todolist-domain/issues"
  },
  "homepage": "https://github.com/douglasduteil/study-multi-target-todolist-domain#readme"
}
