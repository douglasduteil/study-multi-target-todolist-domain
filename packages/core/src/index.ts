//

export * from "./models/todo";
export * from "./interfaces/TodoDataSource";
