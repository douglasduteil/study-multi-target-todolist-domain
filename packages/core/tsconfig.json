{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "composite": true,
    "outDir": "lib",
    "rootDir": "src"
  },
  "exclude": ["**/__test__", "lib", "node_modules"]
}
