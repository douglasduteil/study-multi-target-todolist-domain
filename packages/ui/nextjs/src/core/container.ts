//

import { Container } from "inversify";
import { createContext } from "react";

//

export const container = new Container();
export const ContainerContext = createContext<Container>(container);
