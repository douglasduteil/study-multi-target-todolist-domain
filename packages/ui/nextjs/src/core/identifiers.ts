//

const prefix = "@todolist/packages/ui/nextjs";

export const IDENTIFIER = {
  DATA_SOURCE: Symbol.for(`${prefix}#DATA_SOURCE`)
};
