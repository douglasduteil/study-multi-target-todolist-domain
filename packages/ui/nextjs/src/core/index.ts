//

export * from "./container";
export * from "./identifiers";
