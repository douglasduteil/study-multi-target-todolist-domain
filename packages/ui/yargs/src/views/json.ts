//

import { Todo } from "@todolist/core";

//

export function jsonView(todos: Todo[]): string {
  return JSON.stringify(todos);
}
