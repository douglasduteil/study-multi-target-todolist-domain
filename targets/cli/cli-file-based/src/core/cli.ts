//

import yargs from "yargs";
import { version } from "../../package.json";

export function todolistCli(): yargs.Argv {
  return yargs
    .usage("Usage: $0 <command> [options]")
    .version(version)
    .demandCommand(
      1,
      "A command is required. Pass --help to see all available commands and options."
    )
    .recommendCommands()
    .strict()
    .alias("h", "help")
    .alias("v", "version");
}
