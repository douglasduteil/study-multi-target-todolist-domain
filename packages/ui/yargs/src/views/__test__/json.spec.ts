//

import { Todo } from "@todolist/core";
import { jsonView } from "../json";

//

test("should return list of 0 todos", () => {
  expect(jsonView([])).toMatchSnapshot();
});

test("should return list of one foo todo", () => {
  expect(jsonView([createFakeTodo({ title: "foo" })])).toMatchSnapshot();
});

//

function createFakeTodo(template: Partial<Todo>): Todo {
  return {
    completed: false,
    createdAt: new Date(0),
    id: "",
    title: "",
    updatedAt: new Date(0),
    ...template
  };
}
