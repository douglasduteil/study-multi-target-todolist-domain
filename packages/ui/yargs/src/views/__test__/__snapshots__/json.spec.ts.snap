// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return list of 0 todos 1`] = `"[]"`;

exports[`should return list of one foo todo 1`] = `"[{\\"completed\\":false,\\"createdAt\\":\\"1970-01-01T00:00:00.000Z\\",\\"id\\":\\"\\",\\"title\\":\\"foo\\",\\"updatedAt\\":\\"1970-01-01T00:00:00.000Z\\"}]"`;
